src/test/java/com/mongodb/hibernate/dialect/MongoDialectSettingsTests.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.dialect;
+
+import static java.util.Collections.emptyMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.cfg.JdbcSettings.AUTOCOMMIT;
+import static org.hibernate.cfg.JdbcSettings.JAKARTA_JDBC_URL;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import java.util.Map;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class MongoDialectSettingsTests {
+    @Test
+    void requiresPropertiesThatHaveNoDefaults() {
+        NullPointerException e = assertThrows(NullPointerException.class, () -> MongoDialectSettings.builder(emptyMap())
+                .build());
+        assertThat(e.getMessage()).matches("databaseName must not be null");
```

Comment:
`matches` means RegExp matching. `equals` might be better though for now there is no difference between them.

but imagine if in the future some innocent guy inserted `.` to the end of the message than there might be some subtleties unexpected.

Reply:
nvm. saw the `.matches()` usage later in the same class. It makes sense to use it as it is then.